iOS-ObjC/MSALiOSobjC/ViewController.m

```diff
@@ -0,0 +1,278 @@
+             });
+             [self contentWithToken:[result accessToken]];
+         }];
+        
+    }
+    
+    -(void)acquireTokenInteractively{
+        NSArray *scopes = @[@"https://graph.microsoft.com/user.read"];
+        MSALInteractiveTokenParameters *parameters = [[MSALInteractiveTokenParameters alloc] initWithScopes:scopes];
+        
+        
+        void (^completionBlock)(MSALResult *result, NSError *error) = ^(MSALResult *result, NSError *error) {
+            
+            dispatch_async(dispatch_get_main_queue(), ^{
+                
+                if (result)
+                {
+                    [self updateResultView:result];
+                    dispatch_async(dispatch_get_main_queue(), ^{
+                        _signout.enabled = true ;
+                    });
+                    [self contentWithToken:[result accessToken]];
+                }
+                else
+                {
+                    [self updateResultViewError:error];
+                }
+                
+            });
+        };
+        
+        [application acquireTokenWithParameters:parameters completionBlock:completionBlock];
+    }
+    
+    /**
+     This will invoke the call to the Microsoft Graph API. It uses the
+     built in URLSession to create a connection.
+     */
+    
+    -(void) contentWithToken:(NSString *)token{
+        
+        // Specify the Graph API endpoint
+        NSURL *graphURL = [NSURL URLWithString:(graphURI)];
+        NSMutableURLRequest *graphRequest = [NSMutableURLRequest requestWithURL:graphURL];
+        NSString *authValue = [NSString stringWithFormat:@"Bearer %@", token];
+        // Set the Authorization header for the request. We use Bearer tokens, so we specify Bearer + the token we got from the result
+        [graphRequest addValue:[NSString stringWithFormat:@"Bearer %@", token] forHTTPHeaderField:@"Authorization"];
+        
+        
+        NSURLSession *session = [NSURLSession sharedSession];
+        NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:graphRequest
+                                                    completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                                        NSLog(@"%@",data);
+                                                        if (error)
+                                                        [self updateResultViewError:error];
+                                                        else {
+                                                            NSDictionary *json  = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+                                                            NSLog(@"%@",json);
+                                                            dispatch_async(dispatch_get_main_queue(), ^{
+                                                                [self.resultTextView setText:[json description]];
```

Comment:
update all UI elements in a helper method

Reply:
updated